### PR TITLE
fix: deprecation fix

### PR DESCRIPTION
**Pull request by liuyang1520**
### Summary
This PR addresses the following deprecations:
- `NSURLConnection`, dead code, remove all related code
- `UIWindow keyWindow`, no need to change, our code use it for older iOS version support: [Amplitude-iOS/AMPUtils.m at 06afbb8b9b8da89fa1e7fc58c6342400b77a9ca1 · amplitude/Amplitude-iOS](https://github.com/amplitude/Amplitude-iOS/blob/06afbb8b9b8da89fa1e7fc58c6342400b77a9ca1/Sources/Amplitude/AMPUtils.m#L197-L203) 
- `SecTrustGetCertificateAtIndex`, deprecated, need to use `SecTrustCopyCertificateChain`, [Replace deprecated SecTrustGetCertificateAtIndex in iOS 15?](https://stackoverflow.com/questions/68034788/replace-deprecated-sectrustgetcertificateatindex-in-ios-15)
- `SecTrustEvaluate`, deprecated, need to use `SecTrustEvaluateWithError`, example: [Fixed SecTrustEvaluate deprecated warning by ddaddy · Pull Request #90 · acmacalister/jetfire](https://github.com/acmacalister/jetfire/pull/90/files)
- `UIApplication windows`, deprecated, need to use `UIWindowScene.windows` [How to resolve: 'keyWindow' was deprecated in iOS 13.0](https://stackoverflow.com/a/58031897)

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-iOS/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:
No

### CODE REVIEW

File: Sources/Amplitude/AMPURLConnection.h [hunk 1]
```diff
@@ -1,34 +0,0 @@
-//
-//  AMPURLConnection.h
```
Comment (qingzhuozhen) [inline Sources/Amplitude/AMPURLConnection.h:2]: What about the file `AMPURLConnetion.m`?

Comment (liuyang1520) [reply Sources/Amplitude/AMPURLConnection.h:2]: Thanks for reviewing! Good catch, I thought I have removed it and committed the removal (it doesn't show up in my Xcode). Updated.

File: Sources/Amplitude/ISPPinnedNSURLSessionDelegate.m [hunk 2]
```diff
@@ -20,25 +20,30 @@ - (void)URLSession:(NSURLSession *)session didReceiveChallenge:(NSURLAuthenticat
...
 
         // Validate the certificate chain with the device's trust store anyway
         // This *might* give use revocation checking
-        SecTrustEvaluate(serverTrust, &trustResult);
-        if (trustResult == kSecTrustResultUnspecified) {
-
+        BOOL isTrusted = false;
+        if (@available(iOS 12.0, macos 10.14, *)) {
+            CFErrorRef error;
+            BOOL trusted = SecTrustEvaluateWithError(serverTrust, &error);
```
Comment (yuhao900914) [inline Sources/Amplitude/ISPPinnedNSURLSessionDelegate.m:29]: same here.

File: Sources/Amplitude/AMPUtils.m [hunk 3]
```diff
@@ -187,9 +187,11 @@ + (CGFloat)statusBarHeight {
 
 + (UIWindow *)getKeyWindow {
     if (@available(iOS 13.0, *)) {
-        for (UIWindow *window in [[AMPUtils getSharedApplication] windows]) {
```
Comment (yuhao900914) [inline Sources/Amplitude/AMPUtils.m:190]: I think windows is [deprecated in iOS 15](https://www.reddit.com/r/SwiftUI/comments/p9gxw6/windows_was_deprecated_in_ios_15_any_help_on/), but here is for iOS 13.0. 
Do we need to do any changes here?

Comment (liuyang1520) [reply Sources/Amplitude/AMPUtils.m:190]: I think we are good here. There are two `windows`:
- https://developer.apple.com/documentation/uikit/uiapplication/1623104-windows?language=objc, this one deprecates in iOS 15
- https://developer.apple.com/documentation/uikit/uiwindowscene/3198091-windows?language=objc, this one starts from iOS 13, is the recommended one

**Review by qingzhuozhen — APPROVED** [review]

**Review by yuhao900914 — APPROVED** [review]

**Pull request merged**